Gems/ROS2/Code/Include/ROS2/RobotControl/ControlSubscriptionHandler.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "ROS2/Communication/TopicConfiguration.h"
+#include "ROS2/Frame/ROS2FrameComponent.h"
+#include "ROS2/ROS2Bus.h"
+#include "ROS2/Utilities/ROS2Names.h"
+#include <rclcpp/rclcpp.hpp>
+
+namespace ROS2
+{
+    //! Component extension enabling polymorphic use of generics.
+    class IControlSubscriptionHandler
+    {
+    public:
+        //! Interface handling component activation
+        //! Only activated IComponentActivationHandler will receive and process control messages.
+        //! @param entity Activation context for the owning Component - the entity it belongs to.
+        //! @param subscriberConfiguration configuration with topic and qos
+        virtual void Activate(const AZ::Entity* entity, const TopicConfiguration& subscriberConfiguration) = 0;
+        virtual void Deactivate() = 0;
```

Comment:
Should add document here as well
```suggestion
        //! Interface handling component deactivation
        virtual void Deactivate() = 0;
```